png.unitless: PMACC_DEF_IN_NAMESPACE

Compiling with clang 6.0 for CPU fails in our png unitless static asserts on namespaces.

```
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:138: error: 
      reference to 'Selected' is ambiguous
    PMACC_DEF_IN_NAMESPACE_MSG( You_can_not_scale_your_preview_to_laser_without_using_a_laser___change_png_param, fields::laserProfiles, Selected );
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:114:53: note: 
      expanded from macro 'PMACC_DEF_IN_NAMESPACE_MSG'
  PMACC_CASSERT_MSG( pmacc_msg, ((sizeof(pmacc_msg::var))!=(sizeof(pmacc_msg::fallback::var))) );
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:87:30: note: 
      expanded from macro 'PMACC_CASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG((__VA_ARGS__),pmacc_msg)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:66:33: note: 
      expanded from macro 'PMACC_STATIC_ASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG_DO2(pmacc_cond,pmacc_msg,__COUNTER__,typename pmacc::GetStaticAssertInfoType<__VA_ARGS__>::type)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/include/boost/mpl/assert.hpp:436:56: note: expanded from macro
      'BOOST_MPL_ASSERT_MSG'
BOOST_MPL_ASSERT_MSG_IMPL( BOOST_MPL_AUX_PP_COUNTER(), c, msg, types_ ) \
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/usr/include/boost/mpl/assert.hpp:429:39: note: expanded from macro '\
BOOST_MPL_ASSERT_MSG_IMPL'
  ...boost::mpl::assertion_failed<(c)>( BOOST_PP_CAT(mpl_assert_arg,counter)::assert_arg() ) \
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/mpl/assert.hpp:60:58: note: expanded from macro '\
BOOST_MPL_AUX_ASSERT_CONSTANT'
#   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                         ^~~~
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:138: note: 
      candidate found by name lookup is
      'picongpu::You_can_not_scale_your_preview_to_laser_without_using_a_laser___change_png_param::Selected'
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:138: note: 
      candidate found by name lookup is
      'picongpu::You_can_not_scale_your_preview_to_laser_without_using_a_laser___change_png_param::Selected'
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:138: error: 
      unknown type name 'Selected'
  ...fields::laserProfiles, Selected );
                            ^
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:5: error: 
      type name requires a specifier or qualifier
    PMACC_DEF_IN_NAMESPACE_MSG( You_can_not_scale_your_preview_to_laser_without...
    ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:114:3: note: 
      expanded from macro 'PMACC_DEF_IN_NAMESPACE_MSG'
  PMACC_CASSERT_MSG( pmacc_msg, ((sizeof(pmacc_msg::var))!=(sizeof(pmacc_msg::f...
  ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:87:5: note: 
      expanded from macro 'PMACC_CASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG((__VA_ARGS__),pmacc_msg)
    ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:66:5: note: 
      expanded from macro 'PMACC_STATIC_ASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG_DO2(pmacc_cond,pmacc_msg,__COUNTER__,typename pmacc...
    ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:57:9: note: 
      expanded from macro 'PMACC_STATIC_ASSERT_MSG_DO2'
        BOOST_MPL_ASSERT_MSG(pmacc_cond,PMACC_JOIN(pmacc_msg,PMACC_JOIN(_______...
        ^
/usr/include/boost/mpl/assert.hpp:435:48: note: expanded from macro
      'BOOST_MPL_ASSERT_MSG'
#define BOOST_MPL_ASSERT_MSG( c, msg, types_ ) \
                                               ^
/usr/include/boost/mpl/assert.hpp:429:42: note: expanded from macro '\
BOOST_MPL_ASSERT_MSG_IMPL'
        boost::mpl::assertion_failed<(c)>( BOOST_PP_CAT(mpl_assert_arg,counter)...
                                         ^
In file included from /home/axel/src/picongpu/include/picongpu/main.cpp:24:
In file included from /home/axel/src/picongpu/include/picongpu/../picongpu/simulation_defines.hpp:50:
In file included from /home/axel/src/picongpu/include/picongpu/../picongpu/_defaultUnitless.loader:47:
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:5: error: 
      expected ')'
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:114:3: note: 
      expanded from macro 'PMACC_DEF_IN_NAMESPACE_MSG'
  PMACC_CASSERT_MSG( pmacc_msg, ((sizeof(pmacc_msg::var))!=(sizeof(pmacc_msg::f...
  ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:87:5: note: 
      expanded from macro 'PMACC_CASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG((__VA_ARGS__),pmacc_msg)
    ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:66:5: note: 
      expanded from macro 'PMACC_STATIC_ASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG_DO2(pmacc_cond,pmacc_msg,__COUNTER__,typename pmacc...
    ^
note: (skipping 3 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/include/boost/preprocessor/cat.hpp:22:32: note: expanded from macro 'BOOST_PP_CAT'
#    define BOOST_PP_CAT(a, b) BOOST_PP_CAT_I(a, b)
                               ^
/usr/include/boost/preprocessor/cat.hpp:29:34: note: expanded from macro
      'BOOST_PP_CAT_I'
#    define BOOST_PP_CAT_I(a, b) a ## b
                                 ^
<scratch space>:22:1: note: expanded from here
mpl_assert_arg32
^
/home/axel/src/picongpu/include/picongpu/../picongpu/unitless/png.unitless:32:5: note: 
      to match this '('
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:114:3: note: 
      expanded from macro 'PMACC_DEF_IN_NAMESPACE_MSG'
  PMACC_CASSERT_MSG( pmacc_msg, ((sizeof(pmacc_msg::var))!=(sizeof(pmacc_msg::f...
  ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:87:5: note: 
      expanded from macro 'PMACC_CASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG((__VA_ARGS__),pmacc_msg)
    ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:66:5: note: 
      expanded from macro 'PMACC_STATIC_ASSERT_MSG'
    PMACC_STATIC_ASSERT_MSG_DO2(pmacc_cond,pmacc_msg,__COUNTER__,typename pmacc...
    ^
/home/axel/src/picongpu/include/picongpu/../pmacc/static_assert.hpp:57:9: note: 
      expanded from macro 'PMACC_STATIC_ASSERT_MSG_DO2'
        BOOST_MPL_ASSERT_MSG(pmacc_cond,PMACC_JOIN(pmacc_msg,PMACC_JOIN(_______...
        ^
/usr/include/boost/mpl/assert.hpp:435:48: note: expanded from macro
      'BOOST_MPL_ASSERT_MSG'
#define BOOST_MPL_ASSERT_MSG( c, msg, types_ ) \
                                               ^
/usr/include/boost/mpl/assert.hpp:429:42: note: expanded from macro '\
BOOST_MPL_ASSERT_MSG_IMPL'
        boost::mpl::assertion_failed<(c)>( BOOST_PP_CAT(mpl_assert_arg,counter)...
                                         ^
4 errors generated.
CMakeFiles/picongpu.dir/build.make:62: recipe for target 'CMakeFiles/picongpu.dir/main.cpp.o' failed
make[2]: *** [CMakeFiles/picongpu.dir/main.cpp.o] Error 1
CMakeFiles/Makefile2:178: recipe for target 'CMakeFiles/picongpu.dir/all' failed
make[1]: *** [CMakeFiles/picongpu.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/include/picongpu/unitless/png.unitless
+++ b/include/picongpu/unitless/png.unitless
@@ -17,19 +17,36 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-
-
 #pragma once
 
 #include <pmacc/static_assert.hpp>
+#include <type_traits>
+
 
 namespace picongpu
 {
+namespace traits
+{
+    template< typename >
+    struct is_laser_none : std::false_type {};
+
+    template< typename T >
+    struct is_laser_none< fields::laserProfiles::None< T > > : std::true_type {};
+
+    template< typename >
+    struct is_laser_planewave : std::false_type {};
+
+    template< typename T >
+    struct is_laser_planewave< fields::laserProfiles::PlaneWave< T > > : std::true_type {};
+}
     // asserts for wrong user configurations
     //
     // setting 1: Laser
 #if( EM_FIELD_SCALE_CHANNEL1 == 1 || EM_FIELD_SCALE_CHANNEL2 == 1 || EM_FIELD_SCALE_CHANNEL3 == 1 )
-    PMACC_DEF_IN_NAMESPACE_MSG( You_can_not_scale_your_preview_to_laser_without_using_a_laser___change_png_param, fields::laserProfiles, Selected );
+    PMACC_CASSERT_MSG(
+        You_can_not_scale_your_preview_to_laser_without_using_a_laser___change_png_param,
+        !traits::is_laser_none< fields::laserProfiles::Selected >::value
+    );
 #endif
 
     // setting 2: Drifting Plasma
@@ -51,6 +68,13 @@ namespace picongpu
     // setting 5: Blow Out
 #if( EM_FIELD_SCALE_CHANNEL1 == 5 || EM_FIELD_SCALE_CHANNEL2 == 5 || EM_FIELD_SCALE_CHANNEL3 == 5 )
     //PMACC_CASSERT_MSG( You_can_not_scale_your_preview_to_a_zero_plasma_density___change_png_param, (BASE_DENSITY>0.0) );
-    PMACC_DEF_IN_NAMESPACE_MSG( You_can_not_scale_your_preview_to_blowout_with_a_laser_without_beam_waist___change_png_param, fields::laserProfiles, Selected );
+    PMACC_CASSERT_MSG(
+        You_can_not_scale_your_preview_to_blowout_without_a_laser___change_png_param,
+        !traits::is_laser_none< fields::laserProfiles::Selected >::value
+    );
+    PMACC_CASSERT_MSG(
+        You_can_not_scale_your_preview_to_blowout_with_a_laser_without_beam_waist___change_png_param,
+        !traits::is_laser_planewave< fields::laserProfiles::Selected >::value
+    );
 #endif
 }

